builtin/providers/aws/resource_aws_instance.go

```diff
@@ -760,6 +760,10 @@ func fetchRootDeviceName(ami string, conn *ec2.EC2) (*string, error) {
 		rootDeviceName = image.BlockDeviceMappings[0].DeviceName
 	}
 
+	if rootDeviceName == nil {
+		return nil, fmt.Errorf("[WARN] Error finding Root Device Name for AMI (%s)", ami)
```

Comment:
> terraform refresh should not error

It does not _error_... but a Launch Configuration will show a `-/+` , since the AMI no longer exists ...